services/src/pro/users/oidc.rs

```diff
@@ -359,29 +534,27 @@ impl OidcRequestDb {
             }?,
         };
 
-        let validity = match token_response.expires_in() {
-            None => Err(OidcError::ResponseFieldError {
-                field: "duration".to_string(),
-                reason: "missing".to_string(),
-            }),
-            Some(x) => Ok(x),
-        }?;
+        let oidc_tokens = Self::decode_token_response(&token_response)?;
+
+        let res = OidcIdentityAttributes {
+            user_claims: external_user_claims,
+            oidc_tokens,
+        };
 
-        Ok((user, Duration::milliseconds(validity.as_millis() as i64))) //TODO: Consider allowing u128 for durations to avoid cast.
+        Ok(res)
     }
 
-    #[cfg(test)]
-    pub(in crate::pro) fn from_oidc_with_static_tokens(value: Oidc) -> Self {
-        OidcRequestDb {
-            issuer: value.issuer.to_string(),
-            client_id: value.client_id.to_string(),
-            client_secret: value.client_secret.clone(),
-            redirect_uri: value.redirect_uri.to_string(),
-            scopes: value.scopes,
-            users: Arc::new(Default::default()),
-            state_function: || CsrfToken::new(SINGLE_STATE.to_string()),
-            nonce_function: || Nonce::new(SINGLE_NONCE.to_string()),
-        }
+    async fn refresh_token(
```

Comment:
shouldn't this be called `access_token` as it return an access token?

Reply:
Correct. I think refresh_access_token describes the entire process. ee8a2fc34b0071a07bfcd86d003e9da7adb57ff2